addon/utils/promise-tools.js

```diff
@@ -0,0 +1,56 @@
+import Ember from 'ember';
+
+const isPromise = function(maybePromise) {
+  if (Ember.PromiseProxyMixin.detect(maybePromise)) {
+    return true;
+  }
+
+  if (maybePromise instanceof Ember.RSVP.Promise) {
+    return true;
+  }
+
+  if (maybePromise != null && typeof maybePromise.then === 'function') {
```

Comment:
Both the PromiseProxy and RSVP.Promise have a function called `then` which makes the other two checks unnecessary. They will both return true here and we aren't doing anything different on the other checks.